addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1816,6 +1816,190 @@ QUnit.test('composer should be focused automatically after clicking on the send
+        name: "General",
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                throw new Error();
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    assert.containsNone(
+        document.body,
+        '.o_MessageList',
+        "should not show message list"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_loadingFailed',
+        "should show loading error message"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadView_loadingFailed .btn',
+        "should show a button to allow user to retry loading"
+    );
+});
+
+QUnit.test('failure on loading more messages should display error and prompt retry button', async function (assert) {
+    assert.expect(5);
+
+    let calls = 0;
+    this.data['mail.channel'].records.push({
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        name: "General",
+    });
+    this.data['mail.message'].records.push(...[...Array(60).keys()].map(id => {
+        return {
+            body: 'coucou',
+            channel_ids: [20],
+            id,
+        };
+    }));
+    await this.start({
+        async mockRPC(route, args) {
+            // first call needs to be successful as it is the initial loading of messages
+            // second call comes from load more and needs to fail in order to show the error alert
+            // any later call should work so that retry button and load more clicks would now work
+            if (args.method === 'message_fetch') {
+                calls++;
+                if (calls === 2) {
+                    throw new Error();
+                }
+            }
+            return this._super(...arguments);
+        },
+    });
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        30,
+        "should initially show 30 messages"
+    );
+
+    await afterNextRender(() => document.querySelector('.o_MessageList_loadMore').click());
+    assert.containsN(
+        document.body,
+        '.o_Message',
+        30,
+        "should still show 30 messages as load more has failed"
+    );
```

Comment:
This should be a test on its own, and the assertion is that message display same the same before/after the failed attempt to load more. This should ideally be a single assertion. The part on checking that UI display failed fetch is irrelevant because already covered by another test.